src/sidebar/util/test/root-thread-test.js

```diff
@@ -0,0 +1,273 @@
+import * as annotationFixtures from '../../test/annotation-fixtures';
+import uiConstants from '../../ui-constants';
+import thread from '../root-thread';
+import { $imports } from '../root-thread';
+import immutable from '../immutable';
+
+const fixtures = immutable({
+  emptyThread: {
+    annotation: undefined,
+    children: [],
+  },
+  nonEmptyDraft: {
+    text: 'Some text',
+    tags: [],
+    isPrivate: false,
+  },
+});
+
+describe('sidebar/utils/rootThread', () => {
+  let fakeBuildThread;
+  let fakeFilterAnnotations;
+  let fakeSearchFilter;
+  let fakeThreadState;
+
+  beforeEach(() => {
+    fakeThreadState = {
+      annotations: [],
+      route: 'sidebar',
+      selection: {
+        expanded: {},
+        forcedVisible: [],
+        filters: {},
+        filterQuery: null,
+        selected: [],
+        sortKey: 'Location',
+        selectedTab: uiConstants.TAB_ANNOTATION,
+      },
+    };
+
+    fakeBuildThread = sinon.stub().returns(fixtures.emptyThread);
+    fakeFilterAnnotations = sinon.stub();
+    fakeSearchFilter = {
+      generateFacetedFilter: sinon.stub(),
+    };
+
+    $imports.$mock({
+      './build-thread': fakeBuildThread,
+      './search-filter': fakeSearchFilter,
+      './view-filter': fakeFilterAnnotations,
+    });
+  });
+
+  afterEach(() => {
+    $imports.$restore();
+  });
+
+  describe('#thread', () => {
```

Comment:
```suggestion
  describe('thread', () => {
```

`thread` is definitely a function rather than a method here.